mlshvch/hometask-3/rainfall.rb

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/env ruby
+
+# frozen_string_literal: true
+
+EXIT_COMMAND = 'exit!'
+ERROR_MESSAGE = 'City name can not be blank!'
+
+def extract_data(strng)
+  hash = {}
+  strng.split("\n").each do |data|
+    array = []
+    city, values = data.split(':')
+    results = values.split(',')
+    results.each { |value| array.push(value.split(' ')[1]) }
+    hash[city] = array
+  end
+  hash
```

Comment:
If you need to do any kind of sequential processing and aggregate results in a different object, please consider using [Enumerable#each_with_object](https://devdocs.io/ruby~3/enumerable#method-i-each_with_object) or [Enumerator#with_object](https://devdocs.io/ruby~3/enumerator#method-i-with_object). This way you won't have to declare stuff like `hash = {}`, where the initial state of the hash is never used and variable exists outside of enumeration process. It will also help to eliminate call to `hash` in the end :)

Reply:
I don't understand to to refactor my code using each_with_object

Reply:
I'll give you an example:
```suggestion
  strng.split("\n").each_with_object({}) do |data, city_rainfall_stats|
    array = []
    city, values = data.split(':')
    results = values.split(',')
    results.each { |value| array.push(value.split(' ')[1]) }
    city_rainfall_stats[city] = array
  end
```
What we achieved with such a small change:
1. no out-of-scope variable declaration -> more Rubyish way to do things;
2. `each_with_object` automatically returns the object used as accumulator, in this case it's `city_rainfall_stats`, as a result you don't have to explicitly return it in the end;
3. it's more laconic 🙂

Did it get more clear now? If not, then we can setup a call and talk about it.
Addressing the rest of the comments is up to you, I did nothing else other that demonstrated how you could use `each_with_object` in this particular case.